Source/WebKit/_WebKit_SwiftUI/Foundation+Extras.swift

```
@@ -34,4 +34,18 @@ func onNextMainRunLoop(do body: @escaping @MainActor () -> Void) {
     }
 }
 
+struct AnyEquatable: Equatable {
+    let value: Any
+    private let equals: (Any) -> Bool
+
+    init<E: Equatable>(_ value: E) {
+        self.value = value
+        self.equals = { ($0 as! E) == value }
```

The `as!` seems problematic here. As a type eraser, aren’t you _expecting_ to support mismatched values? This will crash instead of returning `false`.